src/NCronJob/Registry/JobDefinition.cs

```diff
@@ -120,5 +128,57 @@ public RecurringJobSchedule ToRecurringJobSchedule()
             TimeZone: TimeZone!);
     }
 
+    public void UpdateWith(JobOption? jobOption)
+    {
+        if (jobOption is null)
+        {
+            return;
+        }
+
+        if (jobOption.CronExpression is not null)
+        {
+            UserDefinedCronExpression = jobOption.CronExpression;
+            CronExpression = GetCronExpression(jobOption.CronExpression);
+
+            TimeZone = jobOption.TimeZoneInfo;
+        }
+
+        if (jobOption.Parameter is not null)
```

Comment:
We don't want to override the `Parameter` if `null` gets passed in?

Reply:
The only way to pass a `JobOption` in leveraging a parameter is
 - For typed jobs, through the `NCronJobOptionBuilder` .
   => This happens only once per `JobDefinition`. So even considering the possibility to chain `WithParameter()`, we should be safe
 - Through the `RuntimeJobRegistry`.
   Problem is that this is kind of a grey area with regards to expectations. And we're clearly lacking test coverage. Let me see what can be done in this area.

Reply:
>Through the RuntimeJobRegistry.
>Problem is that this is kind of a grey area with regards to expectations. And we're clearly lacking test coverage. Let me see what can be done in this area.

I've added a minor test to ensure we don't later regress and covered the RuntimeRegistry overriding the Parameter with `null` as a value.

I've implemented a slight hack to make this work and added a TODO comment to get rid of it once the NCronJobOptionBuilder is cleaned up for V5.